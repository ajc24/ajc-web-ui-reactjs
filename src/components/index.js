/**
 * Developed by Anthony Cox in 2024
 */

/* Export the header components */
export { default as HeaderTypeA } from './header/HeaderTypeA';

/* Export the main components */
export { default as MainContent } from './main/MainContent';

/* Export the menu bar components */
export { default as MenuBarTypeA } from './menu-bar/MenuBarTypeA';
