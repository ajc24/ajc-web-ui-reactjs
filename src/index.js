/**
 * Developed by Anthony Cox in 2024
 */

/* Import the header components */
import Header from './components/header/Header';

/* Import the image components */
import DecorativeImage from './components/images/DecorativeImage';
import FunctionalImage from './components/images/FunctionalImage';
import InformativeImage from './components/images/InformativeImage';

/* Export all components */
export {
  DecorativeImage,
  FunctionalImage,
  InformativeImage,
  Header,
};
