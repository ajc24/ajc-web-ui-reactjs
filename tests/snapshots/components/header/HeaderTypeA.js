/**
 * Developed by Anthony Cox in 2024
 */
import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-testing-code';
import { HeaderTypeA } from '../../../../src';
import headerBackgroundTestImage from '../../../../stories/images/header-background-test.png';

describe('Header Type A', () => {
  let componentDidMountSpy;

  beforeAll(() => {
    componentDidMountSpy = jest
      .spyOn(HeaderTypeA.prototype, 'componentDidMount')
      .mockImplementation(() => {});
  });

  afterAll(() => {
    componentDidMountSpy.mockRestore();
  });

  describe('Header Type A with no top border (default), transparent background (default), left alignment (default)', () => {
    let snapshot;
    const testComponentId = 'test-header-border-none-bg-default-align-default';

    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HeaderTypeA id={testComponentId}>
            Header text content.
          </HeaderTypeA>
        </React.Fragment>
      );
    });
    
    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Header Type A with black top border (default colour), transparent background, left alignment', () => {
    let snapshot;
    const testComponentId = 'test-header-border-default-bg-transparent-align-left';

    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HeaderTypeA id={testComponentId} showTopBorder={true} backgroundColour="transparent" alignment="left">
            Header text content.
          </HeaderTypeA>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
