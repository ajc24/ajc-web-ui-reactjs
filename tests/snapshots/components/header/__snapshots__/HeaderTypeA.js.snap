// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Type A Header Type A with black top border (default colour), transparent background, left alignment verifies the snapshot for the component 1`] = `"<header role="banner" id="test-header-border-default-bg-transparent-align-left--container" class="ajc-container-screen ajc-header-border-black ajc-background-transparent"><div class="ajc-container-screen-inner"><div id="test-header-border-default-bg-transparent-align-left--content-container" class="ajc-header-content-container ajc-header-content-container-with-border"><div id="test-header-border-default-bg-transparent-align-left--content" class="ajc-header-content ajc-font-default ajc-font-black ajc-header-content-alignment-left">Header text content.</div></div></div></header>"`;

exports[`Header Type A Header Type A with no top border (default), transparent background (default), left alignment (default) verifies the snapshot for the component 1`] = `"<header role="banner" id="test-header-border-none-bg-default-align-default--container" class="ajc-container-screen ajc-background-transparent"><div class="ajc-container-screen-inner"><div id="test-header-border-none-bg-default-align-default--content-container" class="ajc-header-content-container ajc-header-content-container-no-border"><div id="test-header-border-none-bg-default-align-default--content" class="ajc-header-content ajc-font-default ajc-font-black ajc-header-content-alignment-left">Header text content.</div></div></div></header>"`;
